src/test/librados_test_stub/TestClassHandler.cc

```diff
@@ -30,17 +30,30 @@ void TestClassHandler::open_class(const std::string& name,
                                   const std::string& path) {
   void *handle = dlopen(path.c_str(), RTLD_NOW);
   if (handle == NULL) {
-    derr << "Failed to load class: " << dlerror() << dendl;
+    std::cerr << "Failed to load class: " << name << " (" << path << "): "
+              << dlerror() << std::endl;
     return;
   }
-  m_class_handles.push_back(handle);
+
+  // clear any existing error
+  dlerror();
 
   // initialize
   void (*cls_init)() = reinterpret_cast<void (*)()>(
-      dlsym(handle, "__cls_init"));
-  if (cls_init) {
+    dlsym(handle, "__cls_init"));
+
+  char* error = nullptr;
+  if ((error = dlerror()) != nullptr) {
```

Comment:
why do we need to check `dlerror() || cls_init ` instead just checking `cls_init` and print out the error string using `dlerror()` like before?

Reply:
See the man page